src/sentry/backup/comparators.py

```diff
@@ -140,13 +142,40 @@ def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> list[Compa
                 ComparatorFinding(
                     kind=self.get_kind(),
                     on=on,
-                    reason=f"""the left date_updated value on `{on}` ({left_date_updated}) was not
-                            less than or equal to the right ({right_date_updated})""",
+                    reason=f"""the left value ({left_date_updated}) of `{f}` was not less than or equal to the right value ({right_date_updated})""",
                 )
             ]
         return []
 
 
+class DateAddedComparator(JSONScrubbingComparator):
+    """Some exports from before sentry@23.7.1 may trim milliseconds from timestamps if they end in
+    exactly `.000` (ie, not milliseconds at all - what are the odds!). Because comparisons may fail
+    in this case, we use a special comparator for these cases."""
+
+    def __init__(self, *fields: str):
+        super().__init__(*fields)
+
+    def compare(self, on: InstanceID, left: JSONData, right: JSONData) -> list[ComparatorFinding]:
+        findings = []
+        fields = sorted(self.fields)
```

Comment:
what's the reasoning behind sorting here?

Reply:
Discussed offline with Hubert. We actually want the fields to be a set, not a list like they were before this PR, because we want each entry in `self.fields` to be unique. But, we also want the findings for multiple fields to be consistently ordered, otherwise tests are flaky and output is generally unpredictable. If the Python stdlib had something like C++'s `set`, which does enforce ordering, I'd just use that. But it doesn't, so just using a set and ordering at usage time seemed like the best solution.